### ISSUE TITLE
TeamCity Always Fails: Insufficient GPU memory to allocation

### ISSUE THREAD

**Issue by luotao1** [PaddlePaddle/Paddle#5961]
I found TeamCity Always Fails due to **Insufficient GPU memory to allocation**. A PR should be re-run several times to pass the TeamCity check. How should we solve this problem fundamentally?

Following is Conv2dTransposeOp error:
```
[14:32:02] :	 [Step 1/1] ======================================================================
[14:32:02] :	 [Step 1/1] ERROR: test_check_grad_no_filter (__main__.TestConv2dTransposeOp)
[14:32:02] :	 [Step 1/1] ----------------------------------------------------------------------
[14:32:02] :	 [Step 1/1] Traceback (most recent call last):
[14:32:02] :	 [Step 1/1]   File "test_conv2d_transpose_op.py", line 69, in test_check_grad_no_filter
[14:32:02] :	 [Step 1/1]     no_grad_set=set(['Filter']))
[14:32:02] :	 [Step 1/1]   File "/paddle/python/paddle/v2/fluid/tests/op_test.py", line 386, in check_grad
[14:32:02] :	 [Step 1/1]     output_names, no_grad_set)
[14:32:02] :	 [Step 1/1]   File "/paddle/python/paddle/v2/fluid/tests/op_test.py", line 501, in _get_gradient
[14:32:02] :	 [Step 1/1]     for p_name in inputs_with_np for item in inputs_with_np[p_name]
[14:32:02] :	 [Step 1/1]   File "/paddle/python/paddle/v2/fluid/tests/op_test.py", line 501, in <dictcomp>
[14:32:02] :	 [Step 1/1]     for p_name in inputs_with_np for item in inputs_with_np[p_name]
[14:32:02] :	 [Step 1/1]   File "/paddle/python/paddle/v2/fluid/tests/op_test.py", line 424, in _numpy_to_lod_tensor
[14:32:02] :	 [Step 1/1]     tensor.set(np_value, place)
[14:32:02] :	 [Step 1/1] EnforceNotMet: ptr_ should not be null
[14:32:02] :	 [Step 1/1] Insufficient GPU memory to allocation. at [/paddle/paddle/framework/tensor.h:143]
```

**Comment by QiJune** [comment]
I will have a look at this

**Comment by luotao1** [comment]
@QiJune Thanks very much! Every time there's a random op error problem in TeamCity.

**Comment by QiJune** [comment]
In fluid python unit tests, we have get GFLAGS to set fraction of GPU memory usage. But in fluid C++ unit tests, we cannot get GFLAGS.
#5476 solves this, and all fluid C++ unit tests will read a gflags `fraction_of_gpu_memory_to_use`

**Comment by QiJune** [comment]
I make a statistic of job on teamcity, 9 jobs failed because of such error " Insufficient GPU memory to allocation" in among 101 jobs. That's to say about 10% jobs will fail unexpectedly.

**Comment by QiJune** [comment]
Please refer to #6158 , our former GPU memory allocate policy is not reasonable. I have fix it on #6159
